x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -142,11 +142,11 @@ protected static Batch<LogicalPlan> substitutions() {
             new ReplaceAggregateNestedExpressionWithEval(),
             new ReplaceRegexMatch(),
             new ReplaceTrivialTypeConversions(),
+            new ReplaceOrderByExpressionWithEval(),
```

Comment:
Why does this PR move around `ReplaceOrderByExpressionWithEval`? It seems unrelated to this PR's goal.

I just moved this back into its original position and at least the single node spec tests still pass just fine.

If there's no strong reason to move this, can we keep it in the original position? Just so we don't accidentally cause unrelated bugs - and so that git blame doesn't make it look like this was moved into that position for a reason related to INLINESTATS.